app/domain/authentication/authn_k8s/k8s_resource.rb

```diff
@@ -0,0 +1,17 @@
+module Authentication
+  module AuthnK8s
+
+    class K8sResource
+      attr_reader :type, :value
+
+      def initialize(type:, value:)
+        @type = type
+        @value = value
+      end
+
+      def ==(other_resource)
```

Comment:
why an operator and not `matches` here?

Reply:
or `equal` / `isEqual`

Reply:
implementing this operator let's us compare between hashs of AzureResource easily